### PR TITLE
chore: ci-cd-workflow set up with github actions

### PR DESCRIPTION
**Pull request by babul3012**
TISNEW-3848

### CODE REVIEW

File: .aws/task-definition.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "containerDefinitions": [
+    {
+      "name": "tis-trainee-forms",
+      "image": "430723991443.dkr.ecr.eu-west-2.amazonaws.com/tis-trainee-forms:eceec14565f633ce0ccc4ad2cfd812bf3ff469d4",
```
Comment (DorisWongNHS) [inline .aws/task-definition.json:5]: The commit ID at the end may need to be changed

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#server port
+server.port=8207
+#mongodb
+spring.data.mongodb.uri=mongodb://${DB_USER:admin}:${DB_PASSWORD:pwd}@${DB_HOST:localhost}:${DB_PORT:27017}/${DB_NAME:forms}
+#logging
+logging.level.org.springframework.data=debug
+logging.level.root=INFO
+logging.level.uk.nhs.hee.trainee.details=TRACE
```
Comment (DorisWongNHS) [inline src/main/resources/application.properties:8]: Should it be 'trainee.forms' instead of 'trainee.details'?

Comment (babul3012) [reply src/main/resources/application.properties:8]: Yeah, will change it.

File: .aws/task-definition.json (hunk 1 above)
Comment (babul3012) [reply .aws/task-definition.json:5]: Which commit ID?

Comment (DorisWongNHS) [reply .aws/task-definition.json:5]: 'eceec14565f633ce0ccc4ad2cfd812bf3ff469d4' It is an ID for one the commit in trainee-details service (https://github.com/Health-Education-England/tis-trainee-details/commit/eceec14565f633ce0ccc4ad2cfd812bf3ff469d4)

Comment (babul3012) [reply .aws/task-definition.json]: Done

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#server port
```
Comment (Judge40) [inline src/main/resources/application.properties:1]: Move config back to yml format

File: .github/workflows/ci-cd-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        with:
+          task-definition: .aws/task-definition.json
+          container-name: ${{ github.event.repository.name }}
+          image: ${{ steps.login-ecr.outputs.registry }}/${{ github.event.repository.name }}:${{ github.sha }}
+
+      - name: Deploy Amazon ECS task definition
+        uses: aws-actions/amazon-ecs-deploy-task-definition@v1
+        with:
+          task-definition: ${{ steps.task-def.outputs.task-definition }}
+          service: ${{ github.event.repository.name }}
```
Comment (Judge40) [inline .github/workflows/ci-cd-workflow.yml:168]: Service needs to be changed to `trainee-forms`

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+server.port=8207
+#mongodb
+spring.data.mongodb.uri=mongodb://${DB_USER:admin}:${DB_PASSWORD:pwd}@${DB_HOST:localhost}:${DB_PORT:27017}/${DB_NAME:forms}
+#logging
+logging.level.org.springframework.data=debug
+logging.level.root=INFO
+logging.level.uk.nhs.hee.trainee.forms=TRACE
```
Comment (Judge40) [inline src/main/resources/application.properties:8]: Incorrect package name should `uk.nhs.tis.hee.trainee.forms`

**Pull request merged**